jetty9-compat-base/src/main/java/com/google/apphosting/runtime/jetty9/NoOpSessionManager.java

```diff
@@ -0,0 +1,237 @@
+            _idManager = new NoOpSessionIdManager();
+            server.setSessionIdManager(_idManager);
+            server.manage(_idManager);
+            _idManager.start();
+          } finally {
+            Thread.currentThread().setContextClassLoader(oldLoader);
+          }
+        }
+
+        // server session id is never managed by this manager
+        addBean(_idManager, false);
+      }
+    }
+    super.doStart();
+  }
+
+
+  @Override
+  public HttpSession getHttpSession(String id) {
+    return null;
+  }
+
+  @Override
+  public HttpSession newHttpSession(HttpServletRequest request) {
+
+    String id = _idManager.newSessionId(request, System.currentTimeMillis());
+    String nodeId = _idManager.getNodeId(id, request);
+    return new NoOpSession(id, nodeId, _context);
+  }
+
+  @Override
+  public boolean getHttpOnly() {
+    return false;
+  }
+
+  @Override
+  public int getMaxInactiveInterval() {
+    return -1;
+  }
+
+  @Override
+  public void setMaxInactiveInterval(int seconds) {
+  }
+
+  @Override
+  public void setSessionHandler(SessionHandler handler) {
+    _handler = handler;
+  }
+
+  @Override
+  public void addEventListener(EventListener listener) {
+  }
+
+  @Override
+  public void removeEventListener(EventListener listener) {
+
+  }
+
+  @Override
+  public void clearEventListeners() {
+  }
+
+  @Override
+  public HttpCookie getSessionCookie(HttpSession session, String contextPath,
+                                     boolean requestIsSecure) {
+    return null;
+  }
+
+  @Override
+  public SessionIdManager getSessionIdManager() {
+    return _idManager;
+  }
+
+  @Override
+  public SessionIdManager getMetaManager() {
+    return getSessionIdManager();
+  }
+
+  @Override
+  public void setSessionIdManager(SessionIdManager idManager) {
+    _idManager = idManager;
+  }
+
+  @Override
+  public boolean isValid(HttpSession session) {
+    return ((NoOpSession) session).isValid();
+  }
+
+  @Override
+  public String getNodeId(HttpSession session) {
+    return ((NoOpSession) session).getNodeId();
+  }
+
+  @Override
+  public String getClusterId(HttpSession session) {
+    return ((NoOpSession) session).getId();
+  }
+
+  @Override
+  public HttpCookie access(HttpSession session, boolean secure) {
+    return null;
+  }
+
+  @Override
+  public void complete(HttpSession session) {
+    ((NoOpSession) session).complete();
+  }
+
+  @Override
+  public void setSessionIdPathParameterName(String parameterName) {
+  }
+
+  @Override
+  public String getSessionIdPathParameterName() {
+    return null;
+  }
+
+  @Override
+  public String getSessionIdPathParameterNamePrefix() {
+    return null;
+  }
+
+  @Override
+  public boolean isUsingCookies() {
+    return false;
+  }
+
+  @Override
+  public boolean isUsingURLs() {
+    return false;
+  }
+
+  @Override
+  public Set<SessionTrackingMode> getDefaultSessionTrackingModes() {
+    return null;
```

Comment:
The javadoc for the servlet api is silent on this point. See http://docs.oracle.com/javaee/7/api/javax/servlet/ServletContext.html#getEffectiveSessionTrackingModes--